lib/store/index.js

```diff
@@ -10,7 +10,8 @@ import type Kernel from "./../kernel";
 
 class Store {
   subscriptions = new CompositeDisposable();
-  @observable runningKernels = new Map();
+  @observable startingKernels: Map<string, boolean> = new Map();
+  @observable runningKernels: Map<string, Kernel> = new Map();
   @observable editor = atom.workspace.getActiveTextEditor();
```

Comment:
ideally we could do this without even adding `startingKernels`, but we would have to somehow give `store` the new kernel before the kernel is actually ready. I suppose we could promisify or something, but am eager to see what you all think.